### PR TITLE
fix hooked edicts not getting set to FL_EDICT_CHANGED, other tweaks

### PR DESCRIPTION
**Pull request by sapphonie**
on_game_frame addition fixes inconsistent proxy behavior
fix mismatched delete[], thanks valgrind
null init'd all vars
fixed all type cast whining

fixing the inconsistent proxy behavior was hell by the way, i need a drink

### CODE REVIEW

File: extension.cpp [hunk 1]
```diff
@@ -907,7 +907,7 @@ struct callback_t final : prop_reference_t
 
 	void change_edict_state() noexcept
 	{
-		if(ref != INVALID_EHANDLE_INDEX) {
+		if(ref != (int)INVALID_EHANDLE_INDEX) {
```
Comment (arthurdead) [inline extension.cpp]: can you revert this i am gonna fix it later

File: extension.cpp [hunk 2]
```diff
@@ -808,7 +808,7 @@ struct opaque_ptr final
 
 	template <typename T>
 	static void del_hlpr(void *ptr_) noexcept
-	{ delete[] static_cast<T *>(ptr_); }
```
Comment (arthurdead) [inline extension.cpp:811]: can you revert this too

File: extension.cpp [hunk 3]
```diff
@@ -1126,7 +1126,8 @@ struct callback_t final : prop_reference_t
 		if(res == Pl_Changed) {
 			new_pData.emplace<char>(strlen(sp_value)+1);
 			char *new_value{new_pData.get<char>()};
-			strcpy(new_value, new_value);
```
Comment (arthurdead) [inline extension.cpp:1129]: this too

File: extension.cpp (hunk 1 above)
Comment (sapphonie) [reply extension.cpp]: done

File: extension.cpp (hunk 2 above)
Comment (sapphonie) [reply extension.cpp:811]: done

File: extension.cpp (hunk 3 above)
Comment (sapphonie) [reply extension.cpp:1129]: done

**Pull request merged**